kernel/mod_subst.ml

```diff
@@ -294,43 +294,36 @@ let subst_ind sub (ind,i as indi) =
 let subst_pind sub (ind,u) =
   (subst_ind sub ind, u)
 
-let subst_con0 sub (cst,u) =
+let subst_con0 sub cst =
   let mpu,l = Constant.repr2 cst in
   let mpc = KerName.modpath (Constant.canonical cst) in
   let mpu,mpc,resolve,user = subst_dual_mp sub mpu mpc in
   let knu = KerName.make mpu l in
   let knc = if mpu == mpc then knu else KerName.make mpc l in
   match search_delta_inline resolve knu knc with
-    | Some (ctx, t) ->
-      (* In case of inlining, discard the canonical part (cf #2608) *)
```

Comment:
You should probably preserve this comment.

Reply:
Changed.